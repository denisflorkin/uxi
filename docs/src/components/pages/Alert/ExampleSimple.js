import React from 'react';
import Alert from '../../../../../src/Alert';

const ExampleSimple = () => (
  <div>
    <Alert type="danger" noIcon >
      This is an danger Alert without icon and close btn! Daaaanger zone....
    </Alert>
    <br />
    <Alert type="default">
      This is an default Alert with an icon!
    </Alert>
    <br />
    <Alert type="success">
      This is an success Alert with an icon!
    </Alert>
    <br />
    <Alert type="warning">
      This is an warn Alert with an icon!
    </Alert>
    <br />
    <Alert>
      This is an Alert with an icon and close btn!
    </Alert>
    <br />
    <Alert noIcon showClose>
      This is an Alert without icon but with a close btn!
    </Alert>
    <br />
    <Alert>
      This is an monstruous Alert with too much text!
      Sunt reprehenderit incididunt occaecat laborum elit consequat reprehenderit. Velit nulla fugiat adipisicing proident aliqua culpa ex fugiat. Ipsum tempor nostrud amet officia consequat consectetur do labore. Incididunt culpa est irure nisi commodo adipisicing. Quis culpa ullamco est occaecat deserunt culpa occaecat. Officia elit qui ullamco do ipsum mollit aliquip sit excepteur. Elit deserunt magna pariatur deserunt culpa duis minim ex ad proident amet voluptate.
    </Alert>
    <br />
    <Alert showClose>
      This is an monstruous Alert with too much text!
      Sunt reprehenderit incididunt occaecat laborum elit consequat reprehenderit. Velit nulla fugiat adipisicing proident aliqua culpa ex fugiat. Ipsum tempor nostrud amet officia consequat consectetur do labore. Incididunt culpa est irure nisi commodo adipisicing. Quis culpa ullamco est occaecat deserunt culpa occaecat. Officia elit qui ullamco do ipsum mollit aliquip sit excepteur. Elit deserunt magna pariatur deserunt culpa duis minim ex ad proident amet voluptate.
    </Alert>
    <br />
    <Alert isBanner>
      This is a banner Alert!
    </Alert>
    <br />
    <Alert type="danger" isBanner>
      This is a banner Alert!
      Sunt reprehenderit incididunt occaecat laborum elit consequat reprehenderit. Velit nulla fugiat adipisicing proident aliqua culpa ex fugiat. Ipsum tempor nostrud amet officia consequat consectetur do labore. Incididunt culpa est irure nisi commodo adipisicing. Quis culpa ullamco est occaecat deserunt culpa occaecat. Officia elit qui ullamco do ipsum mollit aliquip sit excepteur. Elit deserunt magna pariatur deserunt culpa duis minim ex ad proident amet voluptate.
    </Alert>
    {/* Alert without content has a min width of 50px : */}
    <Alert type="danger" isBanner style={{ left: '76px' }} />

    <Alert type="danger" iconSize={64} >
      I'm an alert with a big icon (custom sized)
    </Alert>
    <h2> Alert with exit handler </h2>
    <Alert type="danger" onClose={() => { window.alert('exited Alert'); }}>
      I'm an alert with a big icon (custom sized)
    </Alert>
  </div>
);

export default ExampleSimple;
