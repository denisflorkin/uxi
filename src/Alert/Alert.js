import React, { Component } from 'react';
import PropTypes from 'prop-types';
import radium from 'radium';
import {
  Close,
  Issue,
  Alert as AlertIcon,
  Done,
} from '../Icons';
import AlertStyle from './Alert.style';
// import EntityIcon from '../Entity/Card/internal/EntityIcon';

class Alert extends Component {
  static propTypes = {
    /**
     * The Alert will be added relativelty to this node.
     */
    children: PropTypes.node,
    /**
     * Show or Hide the
     */
    showClose: PropTypes.bool,
    onClose: PropTypes.func,
    /**
     * One of the value of danger, error, success, warning.
     */
    type: PropTypes.oneOf(['danger', 'error', 'success', 'warning', 'information']),
    noIcon: PropTypes.bool,
    className: PropTypes.string,
    isBanner: PropTypes.bool,
    iconSize: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  };

  static defaultProps = {
    type: 'information',
    showClose: false,
    className: '',
  };

  constructor(props) {
    super(props);

    this.state = {
      isOpen: true,
    };
  }

  /**
   * Close the Alert message when the property showClose is set to 'true'
   */
  close() {
    const { onClose } = this.props;
    if (onClose) {
      onClose();
    }
    this.setState({
      isOpen: false,
    });
  }

  render() {
    const { type, showClose, noIcon, className, isBanner, iconSize, onClose } = this.props;
    const { isOpen } = this.state;

    const wrapperStyles = [AlertStyle.alert];

    if (isBanner) { wrapperStyles.push(AlertStyle.bannerAlert); }

    let closeContent;
    let iconContent;
    let IconComp;
    const mainAlertStyle = [AlertStyle.alertContent];
    if (showClose || isBanner || onClose) {
      closeContent = (
        <button key="closeIcon" onClick={() => { this.close(); }} style={AlertStyle.closeWrapper}>
          <Close color="white" />
        </button>
      );
      mainAlertStyle.push({
        // marginLeft: '15px',
        marginRight: '50px',
      });
    }

    if (type === 'danger' || type === 'error') {
      wrapperStyles.push(AlertStyle.danger);
      IconComp = Issue;
    } else if (type === 'warning') {
      wrapperStyles.push(AlertStyle.warning);
      IconComp = AlertIcon;
    } else if (type === 'success') {
      wrapperStyles.push(AlertStyle.success);
      IconComp = Done;
    } else {
      wrapperStyles.push(AlertStyle.info);
      IconComp = AlertIcon;
    }

    if (!noIcon) {
      iconContent = (
        <IconComp color="white" style={AlertStyle.icon} size={iconSize} />
      );
    }

    if (!isOpen) {
      wrapperStyles.push({
        display: 'none',
      });
    }

    return (
      <div style={wrapperStyles} className={className}>
        {closeContent}
        {iconContent}
        <div style={mainAlertStyle}>
          {this.props.children}
        </div>
      </div>
    );
  }
}

export default radium(Alert);
