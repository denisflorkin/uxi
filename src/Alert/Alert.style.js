export default {
  alert: {
    margin: 0,
    padding: 0,
    boxSizing: 'border-box',
    position: 'relative',
    display: 'flex',
    alignItems: 'center',
  },
  info: {
    color: '#fff',
    backgroundColor: 'rgb(49, 112, 143)',
    borderColor: 'rgb(49, 112, 143)',
  },
  danger: {
    backgroundColor: '#EF5858',
    color: '#fff',
  },
  warning: {
    backgroundColor: '#FF9800',
    color: '#fff',
  },
  success: {
    backgroundColor: '#009688',
    color: '#fff',
  },
  icon: {
    marginLeft: '16px',
    minWidth: '24px',

    // fontSize: '24px',
    // color: '#fff',
    // position: 'absolute',
    // left: '15px',
    // top: '13px',
    // width: '20px',
    // textAlign: 'center',
  },
  alertContent: {
    display: 'inline-block',
    boxSizing: 'border-box',
    padding: '15px',
    // width: '100%',
  },
  closeWrapper: {
    position: 'absolute',
    cursor: 'pointer',
    right: '15px',
    top: '13px',
    ':hover': {
      opacity: '0.6',
    },
    padding: 0,
    margin: 0,
    border: 'none',
    background: 'transparent',
  },
};
