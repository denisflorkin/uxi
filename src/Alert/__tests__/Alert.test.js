import React from 'react';
import Alert from '../Alert';

it('should render an Alert', () => {
  const child = 'This is a message';
  const wrapper = shallow(
    <Alert>{child}</Alert>,
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).toContain(child);
});
