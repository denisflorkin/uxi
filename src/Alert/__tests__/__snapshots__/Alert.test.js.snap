// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an Alert 1`] = `
<styled.div>
  <div
    className=""
    data-radium={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(49, 112, 143)",
        "borderColor": "rgb(49, 112, 143)",
        "boxSizing": "border-box",
        "color": "#fff",
        "display": "flex",
        "margin": 0,
        "minHeight": "50px",
        "padding": 0,
        "position": "relative",
      }
    }
  >
    <Alert
      color="white"
      key=".1"
      style={
        Object {
          "margin": "8px 8px 8px 16px",
          "minWidth": "24px",
        }
      }
    />
    <div
      data-radium={true}
      key=".2"
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "padding": "15px",
          "width": "100%",
        }
      }
    >
      This is a message
    </div>
  </div>
</styled.div>
`;
